tests/test_brightStarStamps.py

```diff
@@ -0,0 +1,143 @@
+            self.assertMaskedImagesAlmostEqual(mi1, mi2)
+        np.testing.assert_almost_equal(self.bss.getMagnitudes(), bss2.getMagnitudes())
+        np.testing.assert_almost_equal(self.bss.getAnnularFluxes(), bss2.getAnnularFluxes())
+        for id1, id2 in zip(self.bss.getGaiaIds(), bss2.getGaiaIds()):
+            self.assertEqual(id1, id2)
+        os.remove(filename)
+
+    def testMagnitudeSelection(self):
+        brightOnly = self.bss.selectByMag(magMax=7)
+        self.assertEqual(len(brightOnly), 2)
+        self.assertFalse("faint" in brightOnly.getGaiaIds())
+
+        faintOnly = self.bss.selectByMag(magMin=7)
+        self.assertEqual(len(faintOnly), 1)
+        self.assertEqual(faintOnly.getGaiaIds()[0], "faint")
+        brightObj = self.bss[self.faintObjIdx]
+        self.assertMaskedImagesAlmostEqual(brightObj["starStamp"], faintOnly.getMaskedImages()[0])
+
+    def testMissingKeyHandling(self):
+        """Test an exception is raised if a star stamp missing some of the
+        required keys is provided at initialization or given to be appended.
+        """
+        fullStar = self.bss[0]
+        falseStars = [fullStar.copy() for _ in range(4)]
+        falseStars[0].pop("starStamp")
+        falseStars[1].pop("gaiaGMag")
+        falseStars[2].pop("gaiaId")
+        falseStars[3].pop("annularFlux")
+        # test at initialization
+        with self.assertRaises(ValueError):
+            _ = brightStarStamps.BrightStarStamps(falseStars, innerRadius=self.innerRadius,
+                                                  outerRadius=self.outerRadius)
+        # test at appending time
+        for fs in falseStars:
+            with self.assertRaises(ValueError):
+                self.bss.append(fs)
+
+    def testAnnulusMismatch(self):
+        """Test an exception is raised if mismatching annulus radii are
+        given (as the annularFlux values would then be meaningless)."""
+        metadata = self.bss.getMetadata()
+        # metadata contains annulus definition; brightStarStamps can be
+        # instanciated without specifying annulus radii
+        _ = brightStarStamps.BrightStarStamps(self.starStamps, metadata=metadata)
+        # Both can be provided as long as values match
+        _ = brightStarStamps.BrightStarStamps(self.starStamps, innerRadius=self.innerRadius,
+                                              outerRadius=self.outerRadius,
+                                              metadata=metadata)
+        # An exception should be raised if they do not
+        with self.assertRaises(AttributeError):
+            _ = brightStarStamps.BrightStarStamps(self.starStamps, innerRadius=self.innerRadius/2,
+                                                  metadata=metadata)
+        with self.assertRaises(AttributeError):
+            _ = brightStarStamps.BrightStarStamps(self.starStamps, outerRadius=self.outerRadius + 1,
+                                                  metadata=metadata)
```

Comment:
What about testing supplying the radius if it does not exist in the metadata, which is probably the only time the arguments will be used.